contributing/ci-release.rst

```diff
@@ -91,7 +91,7 @@ OMERO
         This job builds the OMERO downloads page
 
         #. Checks out the `develop` branch of
-           https://github.com/openmicroscopy/ome-release.git
```

Comment:
This is archived and should likely just be removed. Ok to have that in a follow-up.

Reply:
Yeah, there are a few archived ones linked, possibly some still useful reference though, was indeed trying to keep the edits and judgment calls light for this PR.